cpp/src/arrow/flight/sql/odbc/odbc_api.cc

```diff
@@ -223,4 +243,120 @@ SQLRETURN SQLSetEnvAttr(SQLHENV env, SQLINTEGER attr, SQLPOINTER valuePtr,
+
+    // TODO: Implement SQL_DRIVER_COMPLETE_REQUIRED in SQLDriverConnectW according to the
+    // spec https://github.com/apache/arrow/issues/46448
+#if defined _WIN32 || defined _WIN64
+    if (driverCompletion == SQL_DRIVER_PROMPT ||
+        ((driverCompletion == SQL_DRIVER_COMPLETE ||
+          driverCompletion == SQL_DRIVER_COMPLETE_REQUIRED) &&
+         !missing_properties.empty())) {
+      // TODO: implement driverCompletion behavior to display connection window.
+    }
+#endif
+
+    connection->connect(dsn, properties, missing_properties);
+
+    // Copy connection string to outConnectionString after connection attempt
+    return ODBC::GetStringAttribute(true, connection_string, true, outConnectionString,
+                                    outConnectionStringBufferLen, outConnectionStringLen,
+                                    connection->GetDiagnostics());
+  });
+}
+
+SQLRETURN SQLConnectW(SQLHDBC conn, SQLWCHAR* dsnName, SQLSMALLINT dsnNameLen,
+                      SQLWCHAR* userName, SQLSMALLINT userNameLen, SQLWCHAR* password,
+                      SQLSMALLINT passwordLen) {
+  std::cout << "-AL- SQLConnectW called\n";
+  using driver::flight_sql::FlightSqlConnection;
+  using driver::flight_sql::config::Configuration;
+  using ODBC::ODBCConnection;
+
+  using ODBC::SqlWcharToString;
+
+  ODBCConnection* connection = reinterpret_cast<ODBCConnection*>(conn);
+
+  return ODBCConnection::ExecuteWithDiagnostics(conn, SQL_ERROR, [=]() {
+    std::string dsn = SqlWcharToString(dsnName, dsnNameLen);
+
+    Configuration config;
+    config.LoadDsn(dsn);
+
+    if (userName) {
+      std::string uid = SqlWcharToString(userName, userNameLen);
+      config.Set(FlightSqlConnection::UID, uid);
+    }
+
+    if (password) {
+      std::string pwd = SqlWcharToString(password, passwordLen);
+      config.Set(FlightSqlConnection::PWD, pwd);
+    }
+
+    std::vector<std::string> missing_properties;
+
+    connection->connect(dsn, config.GetProperties(), missing_properties);
+
+    return SQL_SUCCESS;
+  });
+}
+
+SQLRETURN SQLDisconnect(SQLHDBC conn) {
+  std::cout << "-AL- SQLDisconnect called\n";
+  using ODBC::ODBCConnection;
+
+  return ODBCConnection::ExecuteWithDiagnostics(conn, SQL_ERROR, [=]() {
+    ODBCConnection* connection = reinterpret_cast<ODBCConnection*>(conn);
+
+    connection->disconnect();
+
+    return SQL_SUCCESS;
+  });
+}
+
+SQLRETURN SQLGetInfoW(SQLHDBC conn, SQLUSMALLINT infoType, SQLPOINTER infoValuePtr,
+                      SQLSMALLINT bufLen, SQLSMALLINT* length) {
+  std::cout << "-AL- SQLGetInfoW called\n";
+  // TODO: complete implementation of SQLGetInfoW and write tests
+  using ODBC::ODBCConnection;
+
+  return ODBCConnection::ExecuteWithDiagnostics(conn, SQL_ERROR, [=]() {
+    ODBCConnection* connection = reinterpret_cast<ODBCConnection*>(conn);
+
+    // Partially stubbed implementation of SQLGetInfoW
+    if (infoType == SQL_DRIVER_ODBC_VER) {
+      std::string ver("03.80");
```

Comment:
This might need to be conditional based on driver manager. Also, this can be a string_view (and GetStringAttribute should probably be reworked to take in string_views?)

Reply:
I can use `string_view` here but will just keep using `3.80` as it works

Reply:
Updated code to make GetStringAttribute take string_views